[doc][tune] Add tune checkpoint user guide. (#33145)
ray-project/ray: doc/source/tune/key-concepts.rst

> to define your search space.

Reviewer: ```suggestion
to define the search space.
```

When using schedulers, you may face compatibility issues, as shown in the below compatibility matrix.
Certain schedulers cannot be used with search algorithms,

> and certain schedulers require :ref:`checkpointing to be implemented <tune-trial-checkpoint>`.

Reviewer: ```suggestion
and certain schedulers require that you :ref:`implement checkpointing <tune-trial-checkpoint>`.
```

Schedulers can dynamically change trial resource requirements during tuning.

> This is currently implemented in :ref:`ResourceChangingScheduler<tune-resource-changing-scheduler>`,

Reviewer: ```suggestion
This is implemented in :ref:`ResourceChangingScheduler<tune-resource-changing-scheduler>`,
```
